### PR TITLE
TS-41 FEATURE - Open Graph Image

### PR DESCRIPTION
**Pull request by Mark-Ball**
- the purpose of this ticket is just to generate the OG image
- how we actually get this image to twitter, Facebook or wherever is still to come
- could not use the existing Field and Bench for many reasons:
-- we use useContext to access data in those components, and the OG image is generated on the server so i'm pretty sure we don't have access to that
-- React components that are called inside `ImageResponse` cannot be async, which we would need for data fetching
-- the package used under the hood (satori: https://github.com/vercel/satori) does not support a full css implementation, notably it does not support grid
-- we would need to change the existing components anyway because interactivity needed to be removed and the aspect ratio needed to change

to view your OG image, complete a teamsheet and then put the `shareId` into the url, i have not built a link yet
```
http://localhost:3000/api/ogImages?teamSheetId=323811a2-417c-45db-86c8-b8a55227fd14
```

### CODE REVIEW

File: src/app/shared/components/NonInteractiveField/index.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+const NonInteractiveField = (props: NonInteractiveFieldProps) => {
+  const { players, teamSheet } = props
+
+  const data = teamSheet.data as Partial<{ [key: string]: number }>
+
+  return (
+    <div style={{ display: 'flex' }}>
+      <img
+        height="630"
+        src={`${process.env.NEXT_PUBLIC_VERCEL_URL}/rugby_field.svg`}
```
Comment (dcarg) [inline src/app/shared/components/NonInteractiveField/index.tsx:39]: ` ⨯ Error: failed to pipe response
    at pipeToNodeResponse (/Users/david.carr-gregg/code/dcg/teamsheet/node_modules/next/dist/server/pipe-readable.js:106:15)
    at async sendResponse (/Users/david.carr-gregg/code/dcg/teamsheet/node_modules/next/dist/server/send-response.js:41:17)
    at async doRender (/Users/david.carr-gregg/code/dcg/teamsheet/node_modules/next/dist/server/base-server.js:1294:21)
    at async cacheEntry.responseCache.get.incrementalCache.incrementalCache (/Users/david.carr-gregg/code/dcg/teamsheet/node_modules/next/dist/server/base-server.js:1500:28)
    at async DevServer.renderToResponseWithComponentsImpl (/Users/david.carr-gregg/code/dcg/teamsheet/node_modules/next/dist/server/base-server.js:1408:28)
    at async DevServer.renderPageComponent (/Users/david.carr-gregg/code/dcg/teamsheet/node_modules/next/dist/server/base-server.js:1760:24)
    at async DevServer.renderToResponseImpl (/Users/david.carr-gregg/code/dcg/teamsheet/node_modules/next/dist/server/base-server.js:1798:32)
    at async DevServer.pipeImpl (/Users/david.carr-gregg/code/dcg/teamsheet/node_modules/next/dist/server/base-server.js:851:25)
    at async NextNodeServer.handleCatchallRenderRequest (/Users/david.carr-gregg/code/dcg/teamsheet/node_modules/next/dist/server/next-server.js:266:17)
    at async DevServer.handleRequestImpl (/Users/david.carr-gregg/code/dcg/teamsheet/node_modules/next/dist/server/base-server.js:748:17) {
  [cause]: Error: Image source must be an absolute URL: undefined/rugby_field.svg`

Comment (dcarg) [reply src/app/shared/components/NonInteractiveField/index.tsx:39]: { [cause]: Error: Image source must be an absolute URL: undefined/rugby_field.svg

Comment (Mark-Ball) [reply src/app/shared/components/NonInteractiveField/index.tsx:39]: probs caused by a missing env var
```
NEXT_PUBLIC_VERCEL_URL=http://localhost:3000
```

**Comment by dcarg** [comment]
Ok I updated the ogImage for the share link with some positioning and styling.

I've reduced the width and increased the height with view that this will predominantly be viewed on a mobile screen but still not sure what the ideal image size is for this.

Screenshot 2024-01-07 at 5 56 29 pm

File: src/app/shared/components/NonInteractiveField/index.tsx (hunk 1 above)
Comment (dcarg) [reply src/app/shared/components/NonInteractiveField/index.tsx:39]: 🤦

**Comment by dcarg** [comment]
Furthermore have added the TeamSheet Title.

Screenshot 2024-01-07 at 6 26 04 pm

**Comment by dcarg** [comment]
> Furthermore have added the TeamSheet Title.
> 

I've implemented single line elipsis but really wanted to achieve two line elipsis which I can't seem to get to

If you add a maxHeight 60px and remove whiteSpace: 'nowrap', I can get it to only show 2 lines of text and whilst it doesn't elipsis it just won't show a third line if there is one... lol
              
was trying to use webkit-box and lineClamp: 2 but no cigar on a 2 line elipsis yet if you want to have a go.

**Review by dcarg — APPROVED** [review]
Noice

**Pull request merged**